test-network-function/diagnostic/suite.go

```diff
@@ -185,14 +186,22 @@ func getFirstNode(labelFilter map[string]*string) string {
 	return nodeNames[0]
 }
 
-func getMasterNodeName() string {
-	const masterNodeLabel = "node-role.kubernetes.io/master"
-	return getFirstNode(map[string]*string{masterNodeLabel: nil})
+func getMasterNodeName(env *config.TestEnvironment) string {
+	for _, node := range env.Nodes {
```

Comment:
in here, you can start from the back of the map. 
We should try to avoid returning the same node in master and worker.